api/src/test/java/com/epam/pipeline/manager/docker/DockerRegistryNotificationTest.java

```diff
@@ -16,305 +16,374 @@
-        aclTestDao.createAclEntry(registryAclEntry3);
-        registryAclEntry3.setSid(testUserSid);
-        registryAclEntry3.setOrder(2);
-        aclTestDao.createAclEntry(registryAclEntry3);
+    @Autowired
+    private UserManager mockUserManager;
 
-        MockitoAnnotations.initMocks(this);
+    @InjectMocks
+    private final DockerRegistryManager registryManager = new DockerRegistryManager();
 
-        TestUtils.configureDockerClientMock(dockerClient, dockerClientFactoryMock);
+    private DockerRegistry registry1;
+    private DockerRegistry registry2;
+    private DockerRegistry registry3;
+    private ToolGroup toolGroup1;
+    private ToolGroup toolGroup2;
+    private ToolGroup toolGroup3;
+    private Tool tool1;
+    private Tool tool2;
+    private Tool tool3;
 
-        dockerClientFactoryMock.setObjectMapper(new ObjectMapper());
+    @Before
+    public void setUp() {
+        registry1 = getDockerRegistry(REG_ID_1, REPO, USER, "");
+        registry2 = getDockerRegistry(REG_ID_2, REPO_EXTERNAL_PATH, EXTERNAL_PATH, ADMIN);
+        registry3 = getDockerRegistry(REG_ID_3, REPO_NO_WRITE_ACCESS, ADMIN, "");
 
-        when(dockerClientFactoryMock.getDockerClient(any(DockerRegistry.class)))
-                .thenReturn(dockerClient);
-        when(dockerClient.getImageTags(any(String.class), any(String.class)))
-                .thenReturn(Collections.singletonList("latest"));
+        toolGroup1 = getToolGroup(TOOLGROUP_ID_1, "library", registry1.getId(), USER);
+        toolGroup2 = getToolGroup(null, "library2", registry1.getId(), USER);
+        toolGroup3 = getToolGroup(null, "library", registry2.getId(), USER);
 
-        ToolVersion toolVersion = new ToolVersion();
-        toolVersion.setDigest("test_digest");
-        toolVersion.setSize(DOCKER_SIZE);
-        toolVersion.setVersion("test_version");
-        Mockito.doReturn(toolVersion).when(dockerClient)
-                .getVersionAttributes(Mockito.any(), Mockito.any(), Mockito.any());
+        tool1 = getTool(toolGroup1, IMAGE, registry1, USER);
+        tool2 = getTool(toolGroup2, IMAGE_NEW_GROUP, registry1, USER);
+        tool3 = getTool(toolGroup3, IMAGE, registry2, USER);
 
-        doNothing().when(toolVersionManager).updateOrCreateToolVersion(anyLong(), anyString(), anyString(),
-                any(DockerRegistry.class), any(DockerClient.class));
+        MockitoAnnotations.initMocks(this);
     }
 
     @Test
-    @Transactional(propagation = Propagation.REQUIRES_NEW, rollbackFor = Exception.class)
-    @WithMockUser(username = ADMIN, roles = ADMIN_ROLE)
     public void testEventCanEnableToolInExistingGroupWithRightUserAccess() {
-        DockerRegistryEventEnvelope eventsEnvelope = generateDockerRegistryEvents(
-                Collections.singletonList(TEST_USER),
-                Collections.singletonList(TEST_REPO),
-                Collections.singletonList(TEST_IMAGE),
+        setMockBehaviour1();
```

Comment:
Also, to my mind, it would be more cleaner to avoid `common` methods in tests. (it is a bit hard to understand logic with current approach).
I would prefer to write something like this: (for each test with suitable arguments)
```
mockRegistry(...)
mockToolGroup(...)
mockUserContext(...)
```

Reply:
Done!